HIP-1.md

```diff
@@ -0,0 +1,146 @@
+
+Furthermore, the modularity of sovereign consensus allows Hyperlane to be forward compatible with new security models as they emerge.
+
+### **Tech Spec**
+
+Sovereign consensus is powered by `Interchain Security Modules` or `ISMs`, which specify the rules by which a `recipient` will accept an interchain message.
+
+Message `recipients` should specify their `ISM` via the `interchainSecurityModule()` view function.
+
+The `Inbox` contract, when delivering a message via `Inbox.process()`, calls `recipient.interchainSecurityModule().accept()`, which accepts or rejects the message.
+Relayers can inject arbitrary off-chain data (e.g. validator signatures, zero-knowledge proofs, etc.) into `accept()`, allowing for a wide degree of flexibility in `ISM` design.
+
+Some possible ISMs:
+
+- Multisig: Given a message, specifies a set of signers and a threshold that need to have signed `_root` in order to accept a message. Note that the set and threshold can vary based on message content.
+- Optimistic: Given a message, specifies a set of watchers that can pause the `ISM`. Otherwise, merkle roots are accepted, and after a fraud window has expired messages can be proved against these roots.
+- ZKP: Verifies a zero-knowledge-proof of the origin chain's light client protocol, with a merkle proof of the corresponding Hyperlane Outbox's merkle root.
+- Routing: Routes messages to one or more other `ISMs` according to their content.
+
+Individual `ISM` specifications should follow in future HIPs.
+
+```
+interface IInterchainMessageRecipient {
+    // Returns the recipient's ISM, which checks the validity
+    // of interchain messages.
+    function interchainSecurityModule() external view returns (IInterchainSecurityModule);
+
+    function handle(
+        uint32 _origin,
+        bytes32 _sender,
+        bytes calldata _payload
+    ) external;
+}
+
+interface IInterchainSecurityModule {
+    // Used by relayers to determine which off-chain data to inject, and
+    // how to format it.
+    // The first byte must contain the module "type", which informs the relayer
+    // how to interpret the remaining bytes.
+    function moduleData() external view returns (bytes memory);
+
+    // Called by the Inbox to determine whether the provided root
+    // is valid to verify proofs against.
+    function accept(
+        bytes32 _root,
+        bytes calldata _message,
+        bytes calldata _data
+    ) external returns (bool);
+}
+
+interface IInbox {
+
+    // Called by relayers to deliver messages to the recipient.
+    // Verifies the merkle proof of `_message` against `_root` and
+    // calls `recipient.interchainSecurityModule().accept()` before
+    // calling `recipient.handle()`.
+    function process(
+        bytes32 _root,
+        bytes32[32] calldata _proof,
+        bytes calldata _message,
+        bytes calldata _data
```

Comment:
message vs data is not immediately obvious distinction without context

Reply:
Agree "_data" is a terrible variable name